### DISCUSSION TITLE
How to use a fake KWOK cluster with a custom scheduler ?

### DISCUSSION THREAD

**Discussion by annis-souames** [kubernetes-sigs/kwok#1031] [General]
Hello, 

I have setup a fake KWOK cluster with a dozens of nodes and several pods. I would like to alter the scheduler and use a custom scheduler that I built in Python, this scheduler uses a simple round-robin to schedule pods for some research goals. 

Most tutorials mention building the scheduler and then containerize it and deploying it as a pod, then changing the `schedulerName` field in other pods spec, however, I'm not sure that this would work with a simulated cluster using Kwok since all nodes are fake and an actual scheduler cannot be deployed as a pod. 

Is there any way around this ?

**Comment by wzshiming** [comment]
For scheduling KWOK is exactly the same as the real cluster, so maybe you should check the settings for taint and tolerance.
